lucene/test-framework/src/java/org/apache/lucene/store/MockDirectoryWrapper.java

```diff
@@ -745,7 +745,7 @@ public synchronized IndexInput openInput(String name, IOContext context) throws
       maybeThrowDeterministicException();
     }
     if (!LuceneTestCase.slowFileExists(in, name)) {
-      throw randomState.nextBoolean() ? new FileNotFoundException(name + " in dir=" + in) : new NoSuchFileException(name + " in dir=" + in);
+      throw new NoSuchFileException(name + " in dir=" + in);
```

Comment:
Hmm why did we remove the randomness about which (confusingly) different exception to throw here?  This randomness was (is?) useful to help test that Lucene indeed catches `FNFE` and `NSFE` interchangeably.

Reply:
Ops sorry this was work-in-progress and got accidentally pushed (I got too excited from all the tests passed after changing to `LuceneTestCase.newFSDirectory(path)` in test) . 

The reason for the change here is that in `BaseDirectoryTestCase.testPendingDeletions`, it specifically tests for `NoSuchFileException` , so I initially removed it to confirm that's the code failing the test:

https://github.com/apache/lucene-solr/blob/fddb5314fcccf5b3a416362b94a454599b43af84/lucene/test-framework/src/java/org/apache/lucene/store/BaseDirectoryTestCase.java#L1197-L1199

I've reverted this change, and update the test method to check for either exception.

> This randomness was (is?) useful to help test 

I did run `./gradlew check` with the randomness removed a few times, and the tests passed, so either the randomness did not kick in, or the other exception wasn't actually used in code under test.

Reply:
You can "multiply" test runs to avoid re-running manually, Zach. Read up on how to do it (and how it works) here:
https://github.com/apache/lucene-solr/blob/master/help/tests.txt#L84

Reply:
Sorry Dawid for the late reply, I took a week long vacation for Thanksgiving. 

Ah this is a great tip! Was waiting for about half an hour before for each `./gradlew check`. Thanks for the suggestion!